synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -32,6 +32,8 @@
 _REPLACE_STREAM_ORDERING_SQL_COMMANDS = (
     # there should be no leftover rows without a stream_ordering2, but just in case...
     "UPDATE events SET stream_ordering2 = stream_ordering WHERE stream_ordering2 IS NULL",
+    # Build stats on the new column
+    "ANALYZE events(stream_ordering2)"
```

Comment:
the problem with this is that it will happen in the middle of a transaction with all the other commands, and I'm not sure how the locking will interact. I'd suggest doing it in a separate transaction.